packages/datadog-plugin-aws-sdk/src/services/sqs.js

```diff
@@ -134,12 +134,19 @@ class Sqs extends BaseAwsSdkPlugin {
       }
     }
 
-    if (!message.MessageAttributes || !message.MessageAttributes._datadog) return
-
-    const datadogAttribute = message.MessageAttributes._datadog
+    let parsedAttributes = {}
+    if (message.MessageAttributes && message.MessageAttributes._datadog) {
+      const datadogAttribute = message.MessageAttributes._datadog
+      parsedAttributes = this.parseDatadogAttributes(datadogAttribute)
+    }
 
-    const parsedAttributes = this.parseDatadogAttributes(datadogAttribute)
-    if (parsedAttributes) return this.tracer.extract('text_map', parsedAttributes)
+    if (message.Attributes && message.Attributes.AWSTraceHeader) {
```

Comment:
How does this interact with W3C tracestate? We have just started providing guarantees that the tracestate is always kept even if we use our own extraction mechanism. For example, `tracer.extract` will now look at both Datadog and W3C headers to ensure the tracestate is kept. This doesn't seem to be handled here.

In general, I agree with @Qard that this change is significant enough that it shouldn't be done without proper consensus and a specification. Any changes to propagation are extremely difficult to change after the fact since any change would break compatibility with previous versions, including other languages.

Reply:
Follow up question to this; were the same concerns considered for the Java dd-trace-java support for this functionality? (I believe it was this PR: https://github.com/DataDog/dd-trace-java/pull/4730)

Language and platform agnostic interoperability between dd-trace libraries would deliver significant immediate value. 

If we presume that the `tracestate` issue is resolved and the value is persisted, are there other concerns with the AWS-specific logic encapsulated in an SQS-specific implementation?

I'm obviously not the original PR author but I'd like to do what I can to help incorporate this functionality, even if it means pretending I'm a qualified project manager. 😆